cmd/config/config.go

```diff
@@ -644,6 +644,15 @@ func (c Config) SetKVS(s string, defaultKVS map[string]KVS) error {
 		currKVS.Set(Comment, v)
 	}
 
+	hkvs := HelpSubSysMap[subSys]
+	for _, hkv := range hkvs {
+		_, ok := currKVS.Lookup(hkv.Key)
+		if !ok && !hkv.Optional {
```

Comment:
This should be checking for whether value is set, not if key exists
```suggestion
 		v, _ := currKVS.Lookup(hkv.Key)
		if v == "" && !hkv.Optional {
```

Reply:
if the key itself is not present then it means the value is not present @poornas

Reply:
Key seems to be present with empty string as value - just tested cache which expects `drives` as a mandatory key. Current patch does not flag an error when setting an optional key - eg `mc admin config set cache exclude="*.png"`

Reply:
Oh yes you are right key should be present for future requirements, Your requested change is needed.